Polyhedron/demo/Polyhedron/Plugins/PMP/Orient_soup_plugin.cpp

```diff
@@ -254,18 +252,46 @@ void Polyhedron_demo_orient_soup_plugin::createPointsAndPolyline()
   if(item)
   {
     QApplication::setOverrideCursor(Qt::WaitCursor);
-    Scene_points_with_normal_item* points = new Scene_points_with_normal_item();
+    Scene_points_with_normal_item* points = nullptr;
+
+    //remove vertices already in NM edges
+
+    std::vector<std::pair<std::size_t, std::size_t> > to_remove;
```

Comment:
This part seems overly complicated.

Reply:
And I was so proud of it :(

Reply:
Maybe I missed something but to me you only have to collect points that are reported as nm but that are not reported in an edge.

Reply:
Oh but maybe that's the thing. nm edges are only reported using input indices?

Reply:
Yes